### PR TITLE
Save the config as a sorted-key json blob

### PR DESCRIPTION
**Pull request by fahhem**
The tests will likely break, so I'll fix those in a bit.

### CODE REVIEW

File: openhtf/io/output/mfg_inspector.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def _PopulateHeader(record, testrun):
   if 'config' in record.metadata:
     attachment = testrun.info_parameters.add()
     attachment.name = 'config'
-    attachment.value_binary = str(record.metadata['config'])
+    attachment.value_binary = json.dumps(
+        record.metadata['config'], sort_keys=True)
```
Comment (grybmadsci) [inline openhtf/io/output/mfg_inspector.py]: can you add `indent=4` also so it gets pretty-printed?

**Comment by grybmadsci** [comment]
LGTM pending nit inline and this one:

Can you add a comment to the output module somewhere that output callbacks should be deterministic?  It's likely to get overlooked accidentally, but wouldn't hurt to have in there.

**Comment by jettisonjoe** [comment]
Ah, looks like no rebase necessary. Didn't realize how small/easy this fix would be.

LGTM. Will merge once @grybmadsci's nits are addressed.

File: openhtf/io/output/mfg_inspector.py (hunk 1 above)
Comment (fahhem) [inline openhtf/io/output/mfg_inspector.py]: Done

**Comment by jettisonjoe** [comment]
Looks like `record.testrun` needs to be updated with the new indented output.

**Comment by fahhem** [comment]
Fixed and ready to go. Rebased and tested locally, though travis will take its sweet time to catch up

**Comment by grybmadsci** [comment]
@kdsudac pulled this fix into a separate PR that passed tests sooner ;)

**Pull request closed without merging**